docs/OPENSHIFT.md

```diff
@@ -0,0 +1,186 @@
+
+
+## Upgrading
+
+There are two components that can be upgraded independent of each other for each new version of the agent. While not strictly required, we always recommend upgrading both components together.
+
+### Upgrading the Configuration
+
+Not every version update of the agent makes a change to our supplied configuration yamls. These changes will be outlined in our release page to help you determine if you need to update your configuration.
+
+Due to how the agent has evolved over time, certain versions of the agent configuration yamls require different paths to be updated successfully.
+
+If you are unsure of what version of the configuration you have, you can always check the `app.kubernetes.io/version` label of the DaemonSet:
+
+```console
+foo@bar:~$ oc describe daemonset -l app.kubernetes.io/name=logdna-agent
+Name:           logdna-agent
+Selector:       app=logdna-agent
+Node-Selector:  <none>
+Labels:         app.kubernetes.io/instance=logdna-agent
+                app.kubernetes.io/name=logdna-agent
+                app.kubernetes.io/version=2.2.0
+...
+```
+
+Older versions of our configurations do not provide these labels. In that case, each upgrade path belows provides an example of each configuration to be compared to whats running on your cluster.
+
+#### Upgrading from Configuration v1.x.x
+
+* **Example Configuration Yamls:**
+  * [v1.x.x](https://raw.githubusercontent.com/logdna/logdna-agent/master/logdna-agent-ds-os.yaml)
+* **Differences:** This configuration is lacking a number of new OpenShift Objects. It also uses a removed environment variable for controlling journald monitoring, `USEJOURNALD`.
+* **Upgrade Steps:**
+  1. If you have changes you want to persist to the new DaemonSet, backup the old DaemonSet.
+     1. Run `oc get daemonset -o yaml logdna-agent > old-logdna-agent-daemon-set.yaml`.
+     2. Copy any desired changes from `old-logdna-agent-daemon-set.yaml` to the DaemonSet Object in `k8s/agent-resources-openshift.yaml`
+  2. If you want to continue using journald, follow the steps for [enabling journald monitoring on the agent](#enabling-journald-monitoring-on-the-agent).
+  3. Overwrite the DaemonSet as well as create the new OpenShift Objects.
+     1. Run `oc apply -f k8s/agent-resources-openshift.yaml`
+
+> :warning: Exporting OpenShift Objects with "oc get \<resource\> -o yaml" includes extra information about the Object's state. This data does not need to be copied over to the new yaml.
+
+### Upgrading the Image
+
+The image contains the actual agent code that is run on the Pods created by the DaemonSet. New versions of the agent always strive to be backwards compatibility with old configuration versions. Any breaking changes will be outlined on our release page. We always recommend upgrading to the latest configuration to get the best feature support for the agent.
+
```

Comment:
".. always strive to be backwards compatible..." or "always strive for backwards compatibility..."

Reply:
Fix with the latter